tests/ops/pulse/test_convenience_functions.py

```diff
@@ -0,0 +1,426 @@
+
+        constants = np.linspace(0, 9, 10)
+        params = [constants, constants]
+
+        assert qml.math.allclose(circuit(params), true_circuit(params), atol=5e-3)
+        assert qml.math.allclose(
+            jax.grad(circuit)(params), jax.grad(true_circuit)(params), atol=5e-3
+        )
+
+    def test_qnode_pwc_jit(self):
+        """Test that the evolution of a parametrized hamiltonian defined with a pwc function can executed on
+        a QNode using jax-jit, and the results don't differ from execution without jitting."""
+        import jax
+
+        f1 = qml.pulse.pwc((1, 6))
+        f2 = qml.pulse.pwc((0.5, 3))
+        H = f1 * qml.PauliX(0) + f2 * qml.PauliY(1)
+
+        t = (0, 4)
+
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev, interface="jax")
+        def circuit(params):
+            qml.evolve(H)(params=params, t=t)
+            return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+
+        @jax.jit
+        @qml.qnode(dev, interface="jax")
+        def jitted_circuit(params):
+            qml.evolve(H)(params=params, t=t)
+            return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+
+        constants = np.linspace(0, 9, 10)
+        params = [constants, constants]
+
+        assert qml.math.allclose(jitted_circuit(params), circuit(params), atol=5e-3)
+        assert qml.math.allclose(
+            jax.grad(jitted_circuit)(params), jax.grad(circuit)(params), atol=5e-3
+        )
+
+    def test_parametrized_hamiltonian_with_pwc_from_function(self):
+        """Test that a function decorated by pwc_from_function can be used to create a ParametrizedHamiltonian"""
+
+        @qml.pulse.pwc_from_function((2, 5), 20)
+        def f1(params, t):
+            return params + t
+
+        @qml.pulse.pwc_from_function((3, 7), 10)
+        def f2(params, t):
+            return params[0] + params[1] * t**2
+
+        H = f1 * qml.PauliX(0) + f2 * qml.PauliY(1)
+        params = [1.2, [2.3, 3.4]]
+
+        assert isinstance(H, ParametrizedHamiltonian)
+
+        # at t=8 and t=1, both terms are 0
+        assert qml.math.allequal(qml.matrix(H(params, t=8)), 0)
+        assert qml.math.allequal(qml.matrix(H(params, t=1)), 0)
+
+        # at t=2.5, only term 1 is non-zero
+        true_mat = qml.matrix(f1(params[0], 2.5) * qml.PauliX(0), wire_order=[0, 1])
+        assert qml.math.allequal(qml.matrix(H(params, t=2.5)), true_mat)
+
+        # # at t=6, only term 2 is non-zero
+        true_mat = qml.matrix(f2(params[1], 6) * qml.PauliY(1), wire_order=[0, 1])
+        assert qml.math.allequal(qml.matrix(H(params, t=6)), true_mat)
+        #
+        # # at t=4, both are non-zero and output is as expected
+        true_mat = qml.matrix(f1(params[0], 4) * qml.PauliX(0), wire_order=[0, 1]) + qml.matrix(
+            f2(params[1], 4) * qml.PauliY(1), wire_order=[0, 1]
+        )
+        assert qml.math.allequal(qml.matrix(H(params, t=4)), true_mat)
+
+    @pytest.mark.slow
```

Comment:
You can mark it with `@ptest.mark.xfail` as well

Reply:
Passing now